aws/resource_aws_batch_job_queue.go

```diff
@@ -0,0 +1,206 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/batch"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsBatchJobQueue() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBatchJobQueueCreate,
+		Read:   resourceAwsBatchJobQueueRead,
+		Update: resourceAwsBatchJobQueueUpdate,
+		Delete: resourceAwsBatchJobQueueDelete,
+
+		Schema: map[string]*schema.Schema{
+			"compute_environments": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 3,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.JQStateDisabled, batch.JQStateEnabled}, true),
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
Is there any reason why `arn` should be Optional? I don't see it being used anywhere in the CRUD other than in `Set()` which makes me think it should be Computed-only.